lib/ash_authentication/strategies/password/actions.ex

```diff
@@ -38,7 +38,25 @@ defmodule AshAuthentication.Strategy.Password.Actions do
     |> Ash.read()
     |> case do
       {:ok, [user]} ->
-        {:ok, user}
+        case strategy.require_confirmed_with? do
+          nil ->
+            {:ok, user}
+
+          :confirmed_at ->
```

Comment:
Instead of matching on a literal value, this needs to do something like:

```elixir
case strategy.require_confirmed_with do
  nil ->
    {:ok, user}

  field ->
    if Map.get(user, :confirmed_at) != nil do
      ...
    else
      ...
    end
end
```

Additionally, we need to ensure this attribute is selected in the query, so add something like:

```elixir
|> Query.new()
|> Ash.Query.ensure_selected(List.wrap(strategy.require_confirmed_with))
```